presto-orc/src/main/java/com/facebook/presto/orc/OrcWriteValidation.java

```diff
@@ -236,35 +255,146 @@ public void validateRowGroupStatistics(OrcDataSourceId orcDataSourceId, long str
             RowGroupStatistics expectedRowGroup = expectedRowGroupStatistics.get(rowGroupIndex);
             if (expectedRowGroup.getValidationMode() != HASHED) {
                 Map<Integer, ColumnStatistics> expectedStatistics = expectedRowGroup.getColumnStatistics();
-                Set<Integer> actualColumns = actualRowGroupStatistics.keySet().stream()
-                        .map(StreamId::getColumn)
-                        .collect(Collectors.toSet());
-                if (!expectedStatistics.keySet().equals(actualColumns)) {
+                Map<Integer, ColumnStatistics> actualStatistics = aggregateRowGroupStatisticsFromRowIndex(orcDataSourceId, actualRowGroupStatistics, stripeOffset, rowGroupIndex);
+
+                // remove empty row group stats for empty flat maps
+                expectedStatistics = adjustRowGroupStatisticsForFlatMaps(orcDataSourceId, stripeOffset, expectedStatistics);
+                actualStatistics = adjustRowGroupStatisticsForFlatMaps(orcDataSourceId, stripeOffset, actualStatistics);
+
+                Set<Integer> actualColumns = actualStatistics.keySet();
+                Set<Integer> expectedColumns = expectedStatistics.keySet();
+
+                if (!expectedColumns.equals(actualColumns)) {
                     throw new OrcCorruptionException(orcDataSourceId, "Unexpected column in row group %s in stripe at offset %s", rowGroupIndex, stripeOffset);
                 }
+
                 for (Entry<StreamId, List<RowGroupIndex>> entry : actualRowGroupStatistics.entrySet()) {
-                    ColumnStatistics actual = entry.getValue().get(rowGroupIndex).getColumnStatistics();
-                    ColumnStatistics expected = expectedStatistics.get(entry.getKey().getColumn());
+                    int column = entry.getKey().getColumn();
+                    ColumnStatistics actual = actualStatistics.get(column);
+                    ColumnStatistics expected = expectedStatistics.get(column);
                     validateColumnStatisticsEquivalent(orcDataSourceId, "Row group " + rowGroupIndex + " in stripe at offset " + stripeOffset, actual, expected);
                 }
             }
 
             if (expectedRowGroup.getValidationMode() != DETAILED) {
-                RowGroupStatistics actualRowGroup = buildActualRowGroupStatistics(rowGroupIndex, actualRowGroupStatistics);
-                if (expectedRowGroup.getHash() != actualRowGroup.getHash()) {
+                Map<Integer, ColumnStatistics> actualStatistics = aggregateRowGroupStatisticsFromRowIndex(orcDataSourceId, actualRowGroupStatistics, stripeOffset, rowGroupIndex);
+                actualStatistics = adjustRowGroupStatisticsForFlatMaps(orcDataSourceId, stripeOffset, actualStatistics);
```

Comment:
Remove zero-values stats for value nodes for maps containing all null/empty elements. Do this after a bit of cross validation with the stripe column stats.